application-ckeditor-ui/src/main/resources/CKEditor/FileUploader.xml

```diff
@@ -97,6 +98,25 @@
   #end
 #end
 
+## New way of handling attachment without saving immediately the document.
+#macro (handleTemporaryUploadRequest)
+  #if ($services.csrf.isTokenValid($request.form_token))
+    #set ($document = $doc)
+    #if ("$!request.document" != '')
+      #set ($document = $xwiki.getDocument($request.document))
+    #end
+    #set ($reference = $document.documentReference)
+    #set ($attachment = $services.wysiwyg.temporaryUploadAttachment($reference, 'upload'))
+    #if ($attachment)
+      #sendSuccess($document, $attachment.filename)
+    #else
+      #sendError(400 "The 'upload' request parameter is missing. Either you didn't submit any file or the size of the submitted file exceeds the configured maximum upload size.")
```

Comment:
This actually needs to be tested with a maximum upload size since we now throw an exception.